packages/deployments/contracts/contracts_forge/RelayerFeeRouter.t.sol

```diff
@@ -0,0 +1,205 @@
+
+  function setUp() public {
+    home = address(new MockHome());
+    vm.mockCall(xAppConnectionManager, abi.encodeWithSignature("home()"), abi.encode(home));
+    vm.mockCall(xAppConnectionManager, abi.encodeWithSignature("isReplica(address)"), abi.encode(bool(true)));
+    // vm.mockCall(home, abi.encodeWithSelector(Home.dispatch.selector), bytes("0x"));
+    vm.mockCall(home, abi.encodeWithSignature("localDomain()"), abi.encode(localDomain));
+
+    relayerFeeRouterImplementation = new RelayerFeeRouter();
+
+    proxy = new ERC1967Proxy(
+      address(relayerFeeRouterImplementation),
+      abi.encodeWithSelector(RelayerFeeRouter.initialize.selector, xAppConnectionManager, connext)
+    );
+
+    relayerFeeRouter = RelayerFeeRouter(payable(address(proxy)));
+
+    // Shouldn't enrollRemoteRouter be protected by onlyOwner?
+    relayerFeeRouter.enrollRemoteRouter(remoteDomain, bytes32(remote));
+  }
+
+  // ============ initialize ============
+  function testInitializeParameters() public {
+    assertEq(address(relayerFeeRouter.xAppConnectionManager()), address(xAppConnectionManager));
+    assertEq(address(relayerFeeRouter.connext()), address(connext));
+  }
+
+  // ============ setConnext ============
+
+  // Should work
+  function testSetConnext() public {
+    vm.expectEmit(true, true, true, true);
+    emit SetConnext(connext2);
+
+    relayerFeeRouter.setConnext(connext2);
+    assertEq(address(relayerFeeRouter.connext()), connext2);
+  }
+
+  // Fail if not called by owner
+  function testSetConnextOnlyOwner() public {
+    vm.prank(address(0));
+    vm.expectRevert("Ownable: caller is not the owner");
+    relayerFeeRouter.setConnext(connext2);
+  }
+
+  // ============ send ============
+  // Fail if not called by connext
+  function testSendOnlyConnext() public {
+    vm.prank(address(0));
+    vm.expectRevert(abi.encodeWithSelector(RelayerFeeRouter.RelayerFeeRouter__onlyConnext_notConnext.selector));
+
+    bytes32[] memory transactionIds = new bytes32[](2);
+    transactionIds[0] = "AAA";
+    transactionIds[1] = "BBB";
+
+    relayerFeeRouter.send(remoteDomain, recipient, 1, transactionIds);
+  }
+
+  // Fail if not no amount and no transaction ids
+  function testSendClaimEmpty() public {
+    vm.expectRevert(abi.encodeWithSelector(RelayerFeeRouter.RelayerFeeRouter__send_claimEmpty.selector));
+
+    bytes32[] memory transactionIds = new bytes32[](0);
+
+    relayerFeeRouter.send(remoteDomain, recipient, 0, transactionIds);
+  }
+
+  // Fail if empty recipient
+  function testSendRecipientEmpty() public {
+    vm.expectRevert(abi.encodeWithSelector(RelayerFeeRouter.RelayerFeeRouter__send_recipientEmpty.selector));
+
+    bytes32[] memory transactionIds = new bytes32[](2);
+    transactionIds[0] = "AAA";
+    transactionIds[1] = "BBB";
+
+    relayerFeeRouter.send(remoteDomain, address(0), 1, transactionIds);
+  }
+
+  // Should work
+  function testSend(bytes32[] calldata _transactionIds, uint256 _amount) public {
```

Comment:
very nice! been neglecting property-based tests but would be good to start leaning into them